packages/engine/lib/error/src/macros.rs

```diff
@@ -1,62 +1,99 @@
 #[doc(hidden)]
 pub mod __private {
-    #![allow(clippy::unused_self)]
-    //! Uses [autoref-based stable specialization](https://github.com/dtolnay/case-studies/blob/master/autoref-specialization/README.md).
-    // TODO: Expand documentation when string literals are forbidden as this will shrink the
-    //   implementation by a fair bit.
+    //! Implementation detail for macros.
+    //!
+    //! ⚠️ **Functionality in this module is considered unstable and is subject to change at any time
+    //! without a major version bump!** ⚠️
```

Comment:
It's living inside of `pub mod __private`, so it *could* be used. Yes, the documentation will never be rendered and the name of the module is a good suggestion to not use it, but two lines don't hurt I guess. It's required to be public because the macros will use this module.

Reply:
Ah that's unfortunate. Fair enough :)